File: LeetCode/cpp/728.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> ret;
        for (int &num = left; num <= right; ++num) {
            int i = num;
            while (i > 0) {
                int j = i % 10;
                if (j == 0 || num % j != 0)
                    break;
                i /= 10;
            }
            if (i == 0)
                ret.push_back(num);
        }
        return ret;
    }
};

File: LeetCode/python3/728.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        ret = []
        for x in range(left, right + 1):
            y = x
            while y > 0:
                z = y % 10
                if z == 0 or x % z != 0:
                    break;
                y //= 10
            if y == 0:
                ret.append(x)
        return ret